tests/cpp/dlr_pipeline_skl_xgb_test.cc

```diff
@@ -0,0 +1,165 @@
+#include <gtest/gtest.h>
+#include "dlr.h"
+#include "test_utils.hpp"
+
+DLRModelHandle GetDLRModel() {
+  DLRModelHandle model = NULL;
+  const char* model_path0  = "./pipeline_model2/sklearn-preproc";
+  const char* model_path1  = "./pipeline_model2/xgboost-sm";
+  int device_type = 1; // cpu;
+  const char* model_paths[2] = {model_path0, model_path1};
+  if (CreateDLRPipeline(&model, 2 /*count*/, model_paths, device_type, 0) != 0) {
+    LOG(INFO) << DLRGetLastError() << std::endl;
+    throw std::runtime_error("Could not load DLR Model");
+  }
+  return model;
+}
+
+
+TEST(PipelineTest, TestGetDLRNumInputs) {
+  auto model = GetDLRModel();
+  int num_inputs;
+  EXPECT_EQ(GetDLRNumInputs(&model, &num_inputs), 0);
+  EXPECT_EQ(num_inputs, 1);
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestGetDLRNumWeights) {
+  auto model = GetDLRModel();
+  int num_weights;
+  EXPECT_EQ(GetDLRNumWeights(&model, &num_weights), 0);
+  EXPECT_EQ(num_weights, 0);
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestGetDLRInputName) {
+  auto model = GetDLRModel();
+  const char* input_name;
+  EXPECT_EQ(GetDLRInputName(&model, 0, &input_name), 0);
+  EXPECT_STREQ(input_name, "input");
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestGetDLRInputType) {
+  auto model = GetDLRModel();
+  const char* input_type;
+  EXPECT_EQ(GetDLRInputType(&model, 0, &input_type), 0);
+  EXPECT_STREQ(input_type, "json");
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestSetDLRInput) {
+  auto model = GetDLRModel();
+  std::string in_data = "[[0.5,0.6],[0.55,0.66],[0.73,0.83]]";
+  int64_t shape[1] = { (int64_t) in_data.length() };
+  int ndim = 1;
+  const char* input_name = "input";
+  EXPECT_EQ(SetDLRInput(&model, input_name, shape, (void*) in_data.c_str(), ndim), 0);
```

Comment:
Let's use C++ style casts. This one should be `const_cast<char*>(in_data.data())`

Reply:
fixed